groupie/src/main/java/com/genius/groupie/Item.java

```diff
@@ -42,24 +45,53 @@ protected Item(long id) {
         this.id = id;
     }
 
-    public void bind(RecyclerView.ViewHolder viewHolder, int position, List<Object> payloads, OnItemClickListener onItemClickListener) {
+    /**
+     * Perform any actions required to set up the view for display.
+     *
+     * @param holder              The viewholder to bind
+     * @param position            The adapter position
+     * @param payloads            Any payloads (this list may be empty)
+     * @param onItemClickListener An optional adapter-level click listener
+     */
+    @CallSuper
+    public void bind(ViewHolder<T> holder, int position, List<Object> payloads, OnItemClickListener onItemClickListener) {
         this.onItemClickListener = onItemClickListener;
-        ViewHolder<T> holder = (ViewHolder<T>) viewHolder;
-        if (getExtras() != null) {
-            holder.getExtras().putAll(getExtras());
-        }
-        holder.setDragDirs(getDragDirs());
-        holder.setSwipeDirs(getSwipeDirs());
-        if (onItemClickListener != null) {
-            viewHolder.itemView.setOnClickListener(isClickable() ? onClickListener : null);
+        holder.bind(this);
+        if (onItemClickListener != null && isClickable()) {
```

Comment:
I think this changes setting/clearing click listener behavior. Previously, the click listener was cleared if isClickable returns false. Now the click listener is untouched. I think this could leave a click listener still set for a view holder that is being reused.